### PR TITLE
feat(go-feature-flag)!: GO Feature Flag provider rebuild 

### PR DESCRIPTION
**Pull request by thomaspoignant**
## This PR

In this PR the GO Feature Flag provider has been completely rebuilt on top of the OFREP provider.

This implementation adds a bunch of GO Feature Flag features on top of OFREP:
- ✅ Send analytics data to the relay-proxy.
- ✅ Caching _(cache the previous evaluation to reduce the number of API calls)_.
- ✅ Polling _(call an API on the relay-proxy regularly to purge the cache when a configuration change happens on the relay-proxy)_.
- ✅ Eventing _(this version of the provider is supporting OpenFeature eventing)_.

### **⚠️ BREAKING CHANGE**
 This version contains breaking changes on the configuration struct:
- ⚠️ You cannot use the GO module in the provider.  
  _(In order to minimize the dependencies of the GO provider, we have removed the dependency to [`thomaspoignant/go-feature-flag`](https://github.com/thomaspoignant/go-feature-flag), this means that you can't use the GO module with this provider anymore BUT you can now use the new provider for this https://github.com/open-feature/go-sdk-contrib/pull/546)_.
- ⚠️ In the configuration, we removed the interface `HTTPClient` to use directly `*http.Client`.
- ⚠️ To have a fully working cache, you need to have a relay-proxy `v1.32.0` or more.

### Related PR
Fixes #546

### CODE REVIEW

File: release-please-config.json [hunk 1]
```diff
@@ -115,6 +115,14 @@
       "bump-patch-for-minor-pre-major": true,
       "versioning": "default",
       "extra-files": []
+    },
+    "providers/go-feature-flag-in-process": {
```
Comment (beeme1mr) [inline release-please-config.json:119]: Is this supposed to be included in this PR?

Comment (thomaspoignant) [reply release-please-config.json:119]: Actually not, I will remove it

**Comment by mjudeikis** [comment]
Still need to read this code few times more to get my head around but looks healthy
/lgtm

**Comment by toddbaert** [comment]
Will review.

File: providers/go-feature-flag/pkg/controller/goff_api.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	u, _ := url.Parse(g.options.Endpoint)
+	u.Path = path.Join(u.Path, "v1", "flag", "change")
+
+	req, err := http.NewRequest(http.MethodGet, u.String(), nil)
+	if err != nil {
+		return ErrorConfigurationChange, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	if g.options.APIKey != "" {
+		req.Header.Set("Authorization", "Bearer "+g.options.APIKey)
```
Comment (toddbaert) [inline providers/go-feature-flag/pkg/controller/goff_api.go:92]: I'd consider extracting `Authorization`, `Content-Type` etc to some consts.

File: providers/go-feature-flag/pkg/controller/cache.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+func (c *Cache) Purge() {
+	if c.internalCache != nil {
+		c.internalCache.Purge()
+	}
+}
+
+// buildCacheKey builds a cache key from the flag and evaluation context.
+func (c *Cache) buildCacheKey(flag string, evalCtx of.FlattenedContext) uint32 {
+	key := fmt.Sprintf("%s-%+v", flag, evalCtx)
+	h := fnv.New32a()
```
Comment (toddbaert) [inline providers/go-feature-flag/pkg/controller/cache.go:136]: I've never heard of this hash algo, but it seems cool!

Comment (thomaspoignant) [reply providers/go-feature-flag/pkg/controller/cache.go:136]: It is a non crypo hash method, which is perfect for my needs and is faster.

File: providers/go-feature-flag/pkg/controller/http_client.go [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+	"time"
+)
+
+func DefaultHTTPClient() *http.Client {
+	netTransport := &http.Transport{
+		TLSHandshakeTimeout: 10000 * time.Millisecond,
+	}
+
+	return &http.Client{
+		Timeout:   10000 * time.Millisecond,
```
Comment (toddbaert) [inline providers/go-feature-flag/pkg/controller/http_client.go:14]: I guess a KeepAlive here would be good, not sure if by default Go uses one.

**Review by toddbaert — APPROVED** [review]

**Comment by dangra** [comment]
Wanted to point this change causes a panic under certain conditions as explained in #578 and attempted a fix in #579

**Pull request merged**